### PR TITLE
Don't return 1.0 feerate if none is found by `convert_fee_rate`

### LINKED ISSUE #80: Consider dropping `unwrap_or(1.0)` from `convert_fee_rate`
Currently, when something goes wrong/the corresponding `pair` can't be found,  `convert_fee_rate` will default to a bogus `1.0` fee rate:

https://github.com/bitcoindevkit/rust-esplora-client/blob/7faab652bbf5897e0a752354deabedffd22d08aa/src/lib.rs#L90-L94

This however messes with users' assumptions they depend on receiving a correct fee update (e.g. in Lightning). Rather than returning a (btw. undocumented) default value,  `convert_fee_rate` should just error out if something goes wrong, as it's fallible anyways.

### PR DESCRIPTION
**Pull request by ValuedMammal**
Change `convert_fee_rate` to return `Option<f32>` instead of Result.

PR #65 made this function effectively infallible by removing the parse int error while falling back to a bogus 1.0 feerate if a value isn't found in fee estimates. We could make it return an error if for example the given fee estimates map is empty without changing the function signature but in that case we would need a new `Error` variant making it a breaking change anyway, therefore just change the function to return `Option` which should only be `None` if given a target of 0 or an empty map assuming esplora always has a fee estimate for a target of 1 confirmation.

fixes #80

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -88,17 +88,15 @@ pub use r#async::AsyncClient;
...
-    };
-    Ok(fee_val as f32)
+///
+/// Returns `None` if no feerate estimate is found at or below `target` confirmations.
+pub fn convert_fee_rate(target: usize, estimates: HashMap<u16, f64>) -> Option<f32> {
+    let mut pairs = estimates.into_iter().collect::<Vec<(u16, f64)>>();
+    pairs.sort_unstable_by_key(|(k, _)| std::cmp::Reverse(*k));
+    pairs
+        .into_iter()
+        .find(|(k, _)| *k as usize <= target)
```
[the remark this answers is not in the document]
Comment (ValuedMammal) [reply src/lib.rs]: done in ed219e29455a9d38487f50d3862888b73cdb211c

**Review by oleonardolima — APPROVED** [review]
ACK ed219e29455a9d38487f50d3862888b73cdb211c
I think you could add a simple test for it, but fine without it too. https://github.com/bitcoindevkit/rust-esplora-client/blob/ed219e29455a9d38487f50d3862888b73cdb211c/src/lib.rs#L355-L397

**Comment by notmandatory** [comment]
@ValuedMammal this looks good but could you add a simple test as above but also that hits this new `None` case?

**Comment by ValuedMammal** [comment]
I added to the existing `feerate_parsing` test which tests the logic in `convert_fee_rate`. I'm not sure if you're suggesting to add another test? @notmandatory @oleonardolima

**Review by notmandatory — APPROVED** [review]
ACK ec5ee8233011269b2d0bbdf8a1647e27e61b9050

**Comment by notmandatory** [comment]
> I added to the existing `feerate_parsing` test which tests the logic in `convert_fee_rate`. I'm not sure if you're suggesting to add another test? @notmandatory @oleonardolima

Oh sorry I didn't look closely enough at the change you made to that test, that covers it.

**Comment by oleonardolima** [comment]
> I added to the existing `feerate_parsing` test which tests the logic in `convert_fee_rate`. I'm not sure if you're suggesting to add another test? @notmandatory @oleonardolima

Awesome, thanks! You covered the simple test I had in mind in [ec5ee82](https://github.com/bitcoindevkit/rust-esplora-client/pull/90/commits/ec5ee8233011269b2d0bbdf8a1647e27e61b9050)
